demo/scripts/controls/ribbon/RibbonButton.tsx

```diff
@@ -97,36 +103,65 @@ export default class RibbonButton extends React.Component<RibbonButtonProps, Rib
         });
     };
 
-    private renderDropDownItems(
-        items: { [key: string]: string },
-        renderer: DropDownRenderer
-    ): JSX.Element {
-        return (
-            <div className={styles.dropDown}>
-                {Object.keys(items).map(key =>
-                    renderer ? (
-                        <div key={key}>
-                            {renderer(
-                                this.props.plugin.getEditor(),
-                                this.onHideDropDown,
-                                key,
-                                items[key]
-                            )}
-                        </div>
-                    ) : (
-                        <div
-                            key={key}
-                            onClick={() => this.onExecute(key)}
-                            className={styles.dropDownItem}>
-                            {items[key]}
-                        </div>
-                    )
-                )}
-            </div>
-        );
-    }
-
     private getDocument() {
         return this.props.plugin.getEditor().getDocument();
     }
 }
+
+function DropDown(props: {
+    editor: IEditor;
+    button: RibbonButtonType;
+    onHideDropDown: () => void;
+}) {
+    const { editor, button, onHideDropDown } = props;
+    return (
+        <div className={styles.dropDown}>
+            {Object.keys(button.dropDownItems).map(key =>
+                button.dropDownRenderer ? (
+                    <div key={key}>
+                        {button.dropDownRenderer(
+                            editor,
+                            onHideDropDown,
+                            key,
+                            button.dropDownItems[key]
+                        )}
+                    </div>
+                ) : (
+                    <DropDownItem
+                        editor={editor}
+                        itemName={key}
+                        displayName={button.dropDownItems[key]}
+                        buttonOnClick={button.onClick}
+                    />
+                )
+            )}
+        </div>
+    );
+}
+
+function DropDownItem(props: {
+    editor: IEditor;
+    itemName: string;
+    displayName: string;
+    buttonOnClick: (editor: IEditor, key: string) => void;
+}) {
+    const { editor, itemName, displayName, buttonOnClick } = props;
+    const onClick = React.useCallback(() => {
+        editor.stopShadowEdit();
+        buttonOnClick?.(editor, itemName);
+    }, []);
+    const onMouseOver = React.useCallback(() => {
+        editor.startShadowEdit();
+        buttonOnClick?.(editor, itemName);
+    }, []);
+
+    return (
+        <div
+            key={props.itemName}
+            onClick={onClick}
+            onMouseOver={onMouseOver}
+            className={styles.dropDownItem}>
+            {displayName}
```

Comment:
do we need this localized and add aria label?

Reply:
No, this is a demo only.